cli/src/commands/extensions/api.rs

```diff
@@ -0,0 +1,175 @@
+//! Most functions in this module are marked `#[allow(unused)]` for the time being. This
+//! intentional, as there are no clients for those functions in the rest of the code, but
+//! those functions will have to be used by the Deno integration; at that point, we may
+//! remove the annotations.
+
+use std::path::Path;
+use std::str::FromStr;
+
+use crate::commands::parse::{get_packages_from_lockfile, LOCKFILE_PARSERS};
+use crate::config::{get_current_project, Config};
+use crate::{api::PhylumApi, auth::UserInfo};
+
+use anyhow::{anyhow, Context, Error, Result};
+use deno_core::OpState;
+use once_cell::sync::Lazy;
+use phylum_types::types::auth::{AccessToken, RefreshToken};
+use phylum_types::types::common::{JobId, ProjectId};
+use phylum_types::types::job::JobStatusResponse;
+use phylum_types::types::package::{
+    Package, PackageDescriptor, PackageStatusExtended, PackageType,
+};
+use phylum_types::types::project::ProjectDetailsResponse;
+
+/// Container for lazily evaluated dependencies of Extensions API functions. These values won't be
+/// visible from extension code and will have to be set up by the JS runtime builder which will
+/// load their configuration as any other command and then provide the pertinent factories.
+struct InjectedDependencies {
+    api: Lazy<PhylumApi>,
+    config: Lazy<Config>,
+}
+
+impl InjectedDependencies {
+    #[allow(unused)]
+    pub(super) fn from_factories(
```

Comment:
If anything I'd just make it `pub` at that point. The concept of `pub(super)` is just unnecessarily complicated since the relative positioning requires unnecessary cognitive overhead to resolve. So either private, `pub` or `pub(crate)` should be used instead (imo). Overly specific visibility modifiers rarely provide much benefit.

Reply:
I agree that `pub(super)` may be not totally clear and require cognitive overhead, but more general `pub` exposures also go beyond the intended scope of those functions which is very narrow by intentional design (i.e. the Deno runtime should be the only client). Would having those functions be `pub(crate)` and the module being private under `extensions` be a good solution?

Reply:
Yeah that seems fine to me.